common/pubsub/pubsub_async.go

```diff
@@ -0,0 +1,49 @@
+package pubsub
+
+import (
+	"context"
+
+	"github.com/status-im/status-go/common"
+)
+
+// PublishAsync will send the value val of type T on the specified Publisher in a
+// non-blocking manner.
+// If the context is canceled, the value may not be sent to all subscribers.
+func PublishAsync[T any](ctx context.Context, p *Publisher, val T) {
```

Comment:
Just thinking if we could use buffered channels instead of publishing async?

Might be tricky to define the size of buffer, but at the same time with `PublishAsync` there's no limit at all, which also doesn't sound good. If nobody reads the channel, all `PublishAsync` calls become dangling and endlessly stacked. With a buffered channel we'll get both some limit and asynchronicity.

Reply:
> will cause the publisher to be blocked

@dlipicar We can say implement a rate limit. If buffer full, drop messages:
```go
select {
case <-sub.ch:
    // successful write
default:
    // immediately fall here when channel buffer is full
}
```